sklearn/cluster/k_means_.py

```diff
@@ -1225,6 +1233,9 @@ class MiniBatchKMeans(KMeans):
 
     Read more in the :ref:`User Guide <mini_batch_kmeans>`.
 
+    Warning: The data will be converted to C ordering, which might cause a
```

Comment:
I would avoid to issue a warning here. You can add a line of explanation under X, similarly to the some explanation regarding sparse matrix [CSR](http://scikit-learn.org/dev/modules/generated/sklearn.preprocessing.QuantileTransformer.html#sklearn.preprocessing.QuantileTransformer.fit)